extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -144,12 +147,19 @@ public GoogleStorageObjectMetadata getMetadata(
 
   public void delete(final String bucket, final String path) throws IOException
   {
+    // Though currently not documented for the GCS delete api, a false response is indicative of file not found.
+    // All other errors appear as StorageException which is a runtime exceptions. Refer to
+    // https://github.com/googleapis/java-storage/blob/0b5f11af941032e6a55b12d243acf128a6464400/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java#L685
     if (!storage.get().delete(bucket, path)) {
-      throw new IOE(
-          "Failed deleting google cloud storage object [bucket: %s path: %s]",
-          bucket,
-          path
-      );
+      throw new HttpResponseException.Builder(
```

Comment:
it seems weird to construct a http response exception on the client side. Instead, you could throw `java.io.FileNotFoundException` here and in the call below. 

But I am also wondering why throw an exception at all? Other storage connector don't throw such an exception. E.g. I looked into S3 and it behaves in a similar way. It doesn't throw an error if the file doesn't exist.

Reply:
Yeah, I'm also wondering if we need to throw an exception. S3 checks if the index and deprecated descriptor files exist and only then attempt to delete - https://github.com/apache/druid/blob/master/extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3DataSegmentKiller.java#L217. Azure implementation does something similar: https://github.com/apache/druid/pull/15770

I can't seem to find a similar delete-only-if-file-exists pattern in `deleteIfPresent` for google storage

Reply:
I think returning the boolean (directly from the GCS call) will suffice, which can be checked if required. `deleteIfPresent` not checking the return value would implicitly correspond to delete-only-if-file-exists behaviour.

Reply:
Have removed the exception while retaining the void return type, since no delete caller presently has a check for missing files.